corehq/apps/data_analytics/gir_generator.py

```diff
@@ -0,0 +1,134 @@
+import datetime
+
+from collections import namedtuple, defaultdict
+
+from django_countries.data import COUNTRIES
+
+from dimagi.utils.dates import add_months
+
+from corehq.apps.domain.models import Domain
+from corehq.apps.data_analytics.esaccessors import (
+    get_domain_device_breakdown_es, active_mobile_users, get_possibly_experienced
+)
+from corehq.apps.data_analytics.models import MALTRow, GIRRow
+from corehq.apps.data_analytics.const import (
+    TEST_COUCH_TO_SQL_MAP, AMPLIFY_COUCH_TO_SQL_MAP, NOT_SET, BU_MAPPING, NO_BU
+)
+
+
+UserCategories = namedtuple('UserCategories', 'active performing experienced total sms')
+
+
+class GIRTableGenerator(object):
+    """
+        Populates SQL table with data for given list of monthly-datespans
+        See .models.GIRRow
+    """
+
+    def __init__(self, datespan_object_list):
+        self.monthspan_list = datespan_object_list
+
+    def build_table(self):
+        rows = []
+        for domain in Domain.get_all():
+            for monthspan in self.monthspan_list:
+                gir_dict = GIRTableGenerator.get_gir_dict_for_domain_and_monthspan(domain, monthspan)
+                rows.append(gir_dict)
+        GIRRow.objects.bulk_create(
+            [GIRRow(**gir_row) for gir_row in rows]
+        )
+
+    @staticmethod
+    def classify_users(domain, monthspan):
+        performing_users = []
+        experienced_users = []
+        all_users, user_forms, sms = active_mobile_users(domain, monthspan.startdate, monthspan.enddate)
+        user_query = MALTRow.objects.filter(domain_name=domain).values('user_id', 'month').distinct()
+        user_months = defaultdict(int)
+        for entry in user_query:
+            user_months[entry['user_id']] += 1
```

Comment:
I'm not 100% sure this is a problem but my read is that if we run this on a historical period that users future months will count to their experienced threshold, which doesn't seem right. I think to account for that you need to only include months prior to `monthspan` in the original `user_query`